src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -51,32 +52,21 @@ constexpr size_t LSTM_HIDDEN_SIZE = 200;
 constexpr size_t FULLY_CONNECTED_HIDDEN_SIZE = 128;
 constexpr float LSTM_CELL_CLIP_THRESHOLD = 50000.f;
 
-
-// These are the fixed values that the Python implementation currently passes
-// into TCMPS.
-// TODO: These should be exposed in a way that facilitates experimentation.
-// TODO: A struct instead of a map would be nice, too.
-// TODO: And that should really happen before we deploy to Apple platforms where
-//       float and int don't have the same size.
-float_array_map get_training_config(size_t prediction_window,
-                                    int random_seed) {
-  static_assert(sizeof(float) == sizeof(int),
-                "Passing random seed assumes float and int have same size.");
-  float random_seed_float = *reinterpret_cast<float*>(&random_seed);
-
-  return {
-    { "ac_pred_window", shared_float_array::wrap(prediction_window) },
-    { "ac_seq_len", shared_float_array::wrap(NUM_PREDICTIONS_PER_CHUNK) },
-    { "mode", shared_float_array::wrap(0.f) },  // kLowLevelModeTrain
-    { "random_seed", shared_float_array::wrap(random_seed_float) },
-  };
-}
-float_array_map get_inference_config(size_t prediction_window) {
-  return {
-    { "ac_pred_window", shared_float_array::wrap(prediction_window) },
-    { "ac_seq_len", shared_float_array::wrap(NUM_PREDICTIONS_PER_CHUNK) },
-    { "mode", shared_float_array::wrap(1.f) },  // kLowLevelModeInference
-  };
+// Defines the struct needed to create the model backend
+ac_parameters get_parameters(int batch_size, int num_features,
+                             int prediction_window, int num_classes,
+                             int num_predictions_per_chunk, int random_seed,
+                             float mode, float_array_map weights) {
+  ac_parameters ac_params;
+  ac_params.batch_size = batch_size;
+  ac_params.num_features = num_features;
+  ac_params.prediction_window = prediction_window;
+  ac_params.num_classes = num_classes;
+  ac_params.num_predictions_per_chunk = num_predictions_per_chunk;
+  ac_params.random_seed = random_seed;
+  ac_params.mode = mode;
+  ac_params.weights = weights;
+  return ac_params;
```

Comment:
I'm not sure this function adds a lot of value. Might actually be clearer at the call site just to instantiate a `ac_parameters` and set all the parameters explicitly.